circus-train-core/src/test/java/com/hotels/bdp/circustrain/core/transformation/TableParametersTransformationTest.java

```diff
@@ -0,0 +1,146 @@
+/**
+ * Copyright (C) 2016-2019 Expedia Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hotels.bdp.circustrain.core.transformation;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.hadoop.hive.metastore.api.Table;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import com.hotels.bdp.circustrain.api.event.EventTableReplication;
+import com.hotels.bdp.circustrain.core.conf.TableParametersConfig;
+
+@RunWith(MockitoJUnitRunner.class)
+public class TableParametersTransformationTest {
+
+  private static final String KEY = "key";
+  private static final String VALUE = "value";
+  private static final String OVERRIDE_KEY = "override_key";
+  private static final String OVERRIDE_VALUE = "override_value";
+  private static final String SECOND_OVERRIDE_KEY = "second_override_key";
+  private static final String SECOND_OVERRIDE_VALUE = "second_override_value";
+  private static final String EVENT_ID = "event_id";
+  private Table table = new Table();
+  private TableParametersTransformation transformation;
+
+  @Before
+  public void init() {
+    Map<String, String> parameters = new HashMap<>();
+    parameters.put(KEY, VALUE);
+    TableParametersConfig tableParametersConfig = new TableParametersConfig();
+    tableParametersConfig.setParameters(parameters);
+    transformation = new TableParametersTransformation(tableParametersConfig);
+  }
+
+  @Test
+  public void typical() {
+    Table transformedTable = transformation.transform(table);
+    Map<String, String> tableParameters = transformedTable.getParameters();
+    assertThat(tableParameters.get(KEY), is(VALUE));
+  }
+
+  @Test
+  public void typicalImmutableTableParameters() {
+    table.setParameters(Collections.EMPTY_MAP);
+    Table transformedTable = transformation.transform(table);
+    Map<String, String> tableParameters = transformedTable.getParameters();
```

Comment:
Are there any other entries in this returned Map? If not perhaps you want to assert the size of it in this and all the other tests just to ensure there aren't any surprising things being added to it.

Reply:
Good point, added and all green